### PR TITLE
build: Change CMake settings for traccc deps

### PR DESCRIPTION
**Pull request by stephenswat**
This commit enables the build of CUDA components of detray, traccc, and vecmem depending on a new `ACTS_TRACCC_ENABLE_CUDA` flag. It also disables the build of the traccc IO code (which has a circular dependency on ACTS) and cleans up the ACTS CMake configuration. Also adds minor fixes to the ways algebra-plugins and detray are loaded (these are temporary workarounds).

Blocked by:
- https://github.com/acts-project/traccc/pull/704

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -281,6 +281,7 @@ components.
 | ACTS_BUILD_PLUGIN_FPEMON            | Build FPE monitoring plugin<br> type: `bool`, default: `OFF`                                                                                                                                                                       |
 | ACTS_BUILD_PLUGIN_GEOMODEL          | Build GeoModel plugin<br> type: `bool`, default: `OFF`                                                                                                                                                                             |
 | ACTS_BUILD_PLUGIN_TRACCC            | Build Traccc plugin<br> type: `bool`, default: `OFF`                                                                                                                                                                               |
+| ACTS_TRACCC_ENABLE_CUDA             | Enable CUDA for the traccc plugin<br> type: `bool`, default: `OFF`                                                                                                                                                                 |
```
Comment (paulgessinger) [inline docs/getting_started.md:284]:
```suggestion
| ACTS_PLUGIN_TRACCC_ENABLE_CUDA             | Enable CUDA for the traccc plugin<br> type: `bool`, default: `OFF`                                                                                                                                                                 |
```
?

Comment (stephenswat) [reply docs/getting_started.md:284]: I'm open to renaming it; right now it is named in line with the EXATRKX options, e.g. `ACTS_EXATRKX_ENABLE_ONNX`. What do you prefer?

File: CMakeLists.txt [hunk 2]
```diff
@@ -439,14 +440,19 @@ if(ACTS_BUILD_PLUGIN_GEANT4)
 endif()
 
 if(ACTS_BUILD_PLUGIN_TRACCC)
+    # TODO: Algebra-plugins should set this up itself!
+    add_definitions(-DALGEBRA_PLUGINS_INCLUDE_ARRAY)
```
[the remark this answers is not in the document]
Comment (stephenswat) [reply CMakeLists.txt:444]: I decided to fix this in https://github.com/acts-project/traccc/pull/704.

File: docs/getting_started.md (hunk 1 above)
Comment (paulgessinger) [reply docs/getting_started.md:284]: If there's precedent, let's keep it like this.

**Pull request closed without merging**